File: chapter_1/section_2/ride/java/ride.java
/*
ID: mcokzoo1
LANG: JAVA
TASK: ride
*/

import java.io.*;
import java.util.*;

public class ride {
	public static void main(String[] args) throws IOException {
		Scanner in = new Scanner(new File("ride.in"));
		PrintWriter out = new PrintWriter("ride.out");
		int group = 1;
		int comet = 1;
		for (char c : in.next().toCharArray()) {
			group *= (int) c - 64;
		}
		for (char c : in.next().toCharArray()) {
			comet *= (int) c - 64;
		}
		int a = true ? 0 : 1;
		out.println((group-comet) % 47 == 0 ? "GO" : "STAY");
		out.close();
	}
}

File: chapter_1/section_2/ride/python2/ride.py
'''
ID: mcokzoo1
LANG: PYTHON2
TASK: ride
'''

def eval_name(s):
	p = 1
	for c in s:
		p *= ord(c)-64
	return p

if __name__ == '__main__':
	with open('ride.in', 'r') as fin, open('ride.out', 'w') as fout:
		group = fin.readline()
		comet = fin.readline()
		fout.write('GO\n') if (eval_name(group)-eval_name(comet)) % 47 == 0 else fout.write('STAY\n')
